src/Castle.Windsor.Extensions.DependencyInjection/Scope/ExtensionContainerScopeCache.cs

```diff
@@ -19,13 +19,15 @@ namespace Castle.Windsor.Extensions.DependencyInjection.Scope
 
 	internal static class ExtensionContainerScopeCache
 	{
-		internal static readonly AsyncLocal<ExtensionContainerScopeBase> current = new AsyncLocal<ExtensionContainerScopeBase>();
+		private static readonly AsyncLocal<ExtensionContainerScopeBase> current = new AsyncLocal<ExtensionContainerScopeBase>();
 		/// <summary>Current scope for the thread. Initial scope will be set when calling BeginRootScope from a ExtensionContainerRootScope instance.</summary>
 		/// <exception cref="InvalidOperationException">Thrown when there is no scope available.</exception>
 		internal static ExtensionContainerScopeBase Current
 		{
 			get => current.Value ?? throw new InvalidOperationException("No scope available");
 			set => current.Value = value;
 		}
+
+		internal static ExtensionContainerScopeBase TryCurrent => current.Value;
```

Comment:
`TryCurrent` is a strange name, it's not a function. Any reason not to just name it `Current` and just document that is can return null?